molgenis-data-mapper/src/main/java/org/molgenis/data/mapper/controller/MappingServiceController.java

```diff
@@ -650,8 +739,21 @@ public String advancedMappingEditor(@RequestParam(required = true) String mappin
 		AttributeMetaData targetAttribute = targetEntityMetaData != null ? targetEntityMetaData
 				.getAttribute(mappingServiceRequest.getTargetAttributeName()) : null;
 		Repository sourceRepo = dataService.getRepository(mappingServiceRequest.getSourceEntityName());
-		List<Object> calculatedValues = algorithmService.applyAlgorithm(targetAttribute,
+
+		Iterable<AlgorithmEvaluation> algorithmEvaluations = algorithmService.applyAlgorithm(targetAttribute,
```

Comment:
Method should not throw an exception in case of an invalid algorithm (which doesn't happen at the moment). Other RuntimeExceptions should be handled as generic RuntimeExceptions.